Issue while creating a project [Expecting 'ID', got 'INVALID']

### Description
I followed the steps given in the readme.
While running the following command, i am getting this error.

$ /Users/pruthvip/code/test-ms-rnta/node_modules/.bin/plop --plopfile node_modules/react-native-test-app/plopfile.js --dest sample
? What is the name of your test app? lpptestapp
? Which platforms do you need test apps for? all
⠋ [WARN] Cannot find module 'react-native-macos'; skipping macOS target
[WARN] Cannot find module 'react-native-windows'; skipping Windows target
✔  ++ /.watchmanconfig
✖  ++ Parse error on line 71:
...lors.black : Colors.white,          }}>
-----------------------^
Expecting 'ID', got 'INVALID'
✖  ++ app.json Aborted due to previous action failure
✖  ++ babel.config.js Aborted due to previous action failure
✖  ++ index.js Aborted due to previous action failure
✖  ++ metro.config.js Aborted due to previous action failure
✖  ++ package.json Aborted due to previous action failure
✖  ++ react-native.config.js Aborted due to previous action failure
✖  ++ android/gradle/wrapper/gradle-wrapper.jar Aborted due to previous action failure
✖  ++ android/gradle/wrapper/gradle-wrapper.properties Aborted due to previous action failure
✖  ++ android/gradle.properties Aborted due to previous action failure
✖  ++ android/gradlew Aborted due to previous action failure


<!--
  Thank you for taking the time to file this issue.

  Please describe it in detail here:
  - What happened?
  - What did you expect to happen?
  - Can you attach build logs?
  - Can you attach screenshots?
-->

### Your environment

* `react-native-test-app` version:
* Tested platforms:
  * [*] Android
  * [*] iOS
  * [*] macOS
  * [ ] Windows
* System information:
  <!-- Run `npx react-native info` and paste the output here -->
System:
    OS: macOS 10.15.6
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 129.32 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 12.18.3 - ~/.nvm/versions/node/v12.18.3/bin/node
    Yarn: 1.22.5 - /usr/local/bin/yarn
    npm: 6.14.6 - ~/.nvm/versions/node/v12.18.3/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.9.3 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.4, DriverKit 20.2, macOS 11.1, tvOS 14.3, watchOS 7.2
    Android SDK:
      API Levels: 28, 29, 30
      Build Tools: 28.0.3, 29.0.3, 30.0.2
      System Images: android-29 | Google APIs Intel x86 Atom, android-30 | Google APIs Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 4.0 AI-193.6911.18.40.6626763
    Xcode: 12.4/12D4e - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_265 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: ^5.0.1-alpha.2 => 5.0.1-alpha.2 
    react: ^17.0.2 => 17.0.2 
    react-native: ^0.64.0 => 0.64.0 
    react-native-macos: ^0.63.23 => 0.63.23 
  npmGlobalPackages:
    *react-native*: Not Found
### Steps to reproduce

<!--
  Tell us how to reproduce this issue.
  Provide a minimal demo where your issue can be easily reproduced.
-->

--- a/plopfile.js
+++ b/plopfile.js
@@ -11,6 +11,22 @@
  */
 
 /**
+ * Escapes tokens in specified file to prevent Handlebar from attempting to
+ * replace them.
+ *
+ * NOTE: This is a temporary workaround until we can deprecate the Plop template
+ * entirely.
+ *
+ * @param {string} file
+ * @returns {string}
+ */
+function escapeHandlebarTokens(file) {
+  const fs = require("fs");
+  const content = fs.readFileSync(file, { encoding: "utf-8" });
+  return content.replace(/\{\{/g, "\\{{");
+}
+
+/**
  * Returns whether the target platform is included.
  * @param {"all" | Platform} platform
  * @param {Platform} target
@@ -173,7 +189,9 @@ module.exports = (/** @type {import("plop").NodePlopAPI} */ plop) => {
         {
           type: "add",
           path: "App.js",
-          templateFile: require.resolve("react-native/template/App.js"),
+          template: escapeHandlebarTokens(
+            require.resolve("react-native/template/App.js")
+          ),
         },
         {
           type: "add",

